common/dfu_constants.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * dfu_constants.h
+ * Copyright (C) 2015 Simon Newton
+ */
+
+#ifndef COMMON_DFU_CONSTANTS_H_
+#define COMMON_DFU_CONSTANTS_H_
+
+/**
+ * @file dfu_constants.h
+ * @brief Constants for Device Firmware Uploads
+ */
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief DFU message types
+ */
+enum {
+  DFU_DETATCH = 0,
```

Comment:
We have Bootloader/firmware/src/dfu_constants.h but that's going to get removed in an upcoming PR.